### ISSUE TITLE
Continuous "Client heartbeat is timed out, closing connection"

### ISSUE THREAD

**Issue by dimas** [hazelcast/hazelcast#10475]
Hazelcast 3.7.3
after running tests noticed that one of the nodes was continuously logging:
```
2017-05-02T16:20:16.689Z WARN com.hazelcast.client.impl.ClientHeartbeatMonitor Client heartbeat is timed out, closing connection to Connection[id=2996, 0.0.0.0/0.0.0.0:5701->null, endpoint=[10.0.168.74]:52264, alive=false, type=JAVA_CLIENT]. Now: 2017-05-02 16:20:16.689. LastTimePacketReceived: 2017-05-02 15:52:09.671
2017-05-02T16:20:26.689Z WARN com.hazelcast.client.impl.ClientHeartbeatMonitor Client heartbeat is timed out, closing connection to Connection[id=2996, 0.0.0.0/0.0.0.0:5701->null, endpoint=[10.0.168.74]:52264, alive=false, type=JAVA_CLIENT]. Now: 2017-05-02 16:20:26.689. LastTimePacketReceived: 2017-05-02 15:52:09.671
2017-05-02T16:20:36.689Z WARN com.hazelcast.client.impl.ClientHeartbeatMonitor Client heartbeat is timed out, closing connection to Connection[id=2996, 0.0.0.0/0.0.0.0:5701->null, endpoint=[10.0.168.74]:52264, alive=false, type=JAVA_CLIENT]. Now: 2017-05-02 16:20:36.689. LastTimePacketReceived: 2017-05-02 15:52:09.671
2017-05-02T16:20:46.689Z WARN com.hazelcast.client.impl.ClientHeartbeatMonitor Client heartbeat is timed out, closing connection to Connection[id=2996, 0.0.0.0/0.0.0.0:5701->null, endpoint=[10.0.168.74]:52264, alive=false, type=JAVA_CLIENT]. Now: 2017-05-02 16:20:46.689. LastTimePacketReceived: 2017-05-02 15:52:09.671
```
note that it complains about the very same connection over and over. Probably should have closed it the first time it did not like it :)

**Comment by dimas** [comment]
More interesting - it seems that the logging continued after the client was restarted so there were definitely NO connection from the client host on port 52264 to the Hazelcast cluster and yet, Hazelcast kept complaining about it.

**Comment by sancar** [comment]
Hi @dimas , 
This should be fixed in 3.8 and 3.7.7. Can you try those ?

**Comment by dimas** [comment]
Can try 3.7.7 (will take time) but not 3.8.
Feel free to close if you think it is already resolved. If still happens with 3.7.7 I can raise a new one.

**Comment by dimas** [comment]
btw, is that warning a harmless thing or does it mean that node entered some bad state or anything like that? Do I need to restart it because in addition to that warning there is something else that most likely broke ?

**Comment by sancar** [comment]
I found the first report of this here:
https://github.com/hazelcast/hazelcast/pull/9271

First fix is done in 3.7.4
https://github.com/hazelcast/hazelcast/pull/9308

First of all, this causes memory leak obviously. Secondly, if you are using listeners it will cause unnecessary load. There will be two event registrations on server side eventough one of them will not fire events to client. 
I highly recommend you to upgrade to at least 3.7.7.
